app/public/cantusdata/test/core/helpers/mei_processing/test_bounding_box_utils.py

```diff
@@ -0,0 +1,49 @@
+from unittest import TestCase
+from typing import List, Tuple
+from cantusdata.helpers.mei_processing.bounding_box_utils import (
+    combine_bounding_boxes,
+    combine_bounding_boxes_single_system,
+    stringify_bounding_boxes,
+)
+from cantusdata.helpers.mei_processing.mei_parsing_types import Zone
+
+
+class TestBoundingBoxUtils(TestCase):
+
+    def test_combine_bounding_boxes_single_system(self) -> None:
+        bounding_boxes: List[Zone] = [
+            {"coordinates": (0, 0, 1, 1), "rotate": 0},
+            {"coordinates": (1, 1, 2, 2), "rotate": 0},
+        ]
+        combined_box = combine_bounding_boxes_single_system(bounding_boxes)
+        expected_combined_box = {"coordinates": (0, 0, 2, 2), "rotate": 0.0}
+        self.assertEqual(combined_box, expected_combined_box)
+
+    def test_combine_bounding_boxes(self) -> None:
+        bounding_boxes: List[Tuple[Zone, int]] = [
+            ({"coordinates": (0, 0, 1, 1), "rotate": 0.0}, 1),
+            ({"coordinates": (1, 1, 2, 2), "rotate": 0.0}, 1),
+            ({"coordinates": (2, 1, 3, 2), "rotate": 0.0}, 2),
+            ({"coordinates": (3, 3, 4, 4), "rotate": 0.0}, 2),
+            ({"coordinates": (4, 3, 5, 6), "rotate": 0.0}, 2),
+            ({"coordinates": (8, 8, 9, 10), "rotate": 0.0}, 3),
```

Comment:
The last item in each tuple is the system that the bounding box belongs to, correct? And since three distinct values appear in the second position of the tuple, we end up with three different `expected_combined_boxes`?

This could be made clearer by either renaming the test to `test_combine_bounding_boxes_multiple_systems` or adding comments.

Reply:
Or even, would it make sense to create a `BoundingBox` type spelling out the meaning of the two values of this tuple?

Reply:
Also, can I suggest reordering the coordinates of the various boxes? In `combine_bounding_boxes`, you say
```
# Combine the bounding boxes for each system, in system order
combined_boxes = []
for _, boxes in sorted(system_boxes.items()):
    ...
```
but it's not immediately clear to me that `sorted` will return a list sorted in system order rather than ordered by first coordinate or something else. I'm worried that this test is meant to test that the combined bounding boxes be ordered in system order, but it's passing coincidentally since the first items of all the coordinates happen to be in system order too.

Reply:
> This could be made clearer by either renaming the test to test_combine_bounding_boxes_multiple_systems or adding comments.

Ok, I've made a few changes...including by adding subtests; since we want `combine_bounding_boxes` to handle a single-system case correctly as well, it seems like the additional subtest I added makes that clear. Let me know.

> Also, can I suggest reordering the coordinates of the various boxes?

Sure...good point.